File: JavaScript/7js 2/getPositions.js
/*
|--------------------------------------------------------------------------
| Hands Up
|--------------------------------------------------------------------------
|
*/

function getPositions(s) {
   return [s % 3, ((s / 3) | 0) % 3, ((s / 9) | 0) % 3];
}


function getPositions(s) {
   return ("000" + s.toString(3)).slice(-3).split("").map(Number).reverse();
}


function getPositions(s) {
   return [s % 3, ~~(s / 3) % 3, ~~(s / 9) % 3];
}


console.log(getPositions(54), [0, 0, 0]);
console.log(getPositions(98), [2, 2, 1]);
console.log(getPositions(3), [0, 1, 0]);

File: Python/7py 2/get_positions.py
"""
|--------------------------------------------------------------------------
| Hands Up
|--------------------------------------------------------------------------
|
"""

def get_positions(s):
   return (s % 3, (s // 3) % 3, (s // 9) % 3)


def get_positions(n):
   n, x = divmod(n, 3)
   n, y = divmod(n, 3)
   n, z = divmod(n, 3)
   return x, y, z


def get_positions(n):
   return tuple((n // 3**i) % 3 for i in range(3))


print(get_positions(54), (0, 0, 0))
print(get_positions(98), (2, 2, 1))
print(get_positions(3), (0, 1, 0))